tests/integration/security/rbac/v2/basic/extended_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basic
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/security/rbac/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+// TestRBACV2Extended tests extended features of RBAC v2 such as global namespace and inline role def.
+func TestRBACV2Extended(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, "rbacv2-extended-test", true)
+			ports := []echo.Port{
+				{
+					Name:        "http",
+					Protocol:    model.ProtocolHTTP,
+					ServicePort: 80,
+				},
+				{
+					Name:        "tcp",
+					Protocol:    model.ProtocolTCP,
+					ServicePort: 90,
+				},
+			}
+			a := echoboot.NewOrFail(t, ctx, echo.Config{
+				Service:        "a",
+				Namespace:      ns,
+				Sidecar:        true,
+				ServiceAccount: true,
+				Ports:          ports,
+				Galley:         g,
+				Pilot:          p,
+			})
+			b := echoboot.NewOrFail(t, ctx, echo.Config{
+				Service:        "b",
+				Namespace:      ns,
+				Ports:          ports,
+				Sidecar:        true,
+				ServiceAccount: true,
+				Galley:         g,
+				Pilot:          p,
+			})
+			c := echoboot.NewOrFail(t, ctx, echo.Config{
+				Service:        "c",
+				Namespace:      ns,
+				Ports:          ports,
+				Sidecar:        true,
+				ServiceAccount: true,
+				Galley:         g,
+				Pilot:          p,
+			})
+
+			cases := []util.TestCase{
+				{
+					Request: connection.Checker{
```

Comment:
For b->a, can we test the allowed/denied case at the same time? So that you can confirm the ServiceRole in root namespace works for both cases.
The current test case is either all allowed or denied based on `isMtlsEnabled`.

Reply:
Done. Thanks for the suggestion